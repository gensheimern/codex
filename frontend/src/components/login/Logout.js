import React from 'react';
import Snackbar from 'material-ui/Snackbar';

export default class Logout extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			open: false,
		};
	}

	componentDidMount() {
		window.localStorage.removeItem('apiToken');

		this.setState({
			open: true,
		});
	}

	handleRequestClose = () => {
		this.setState({
			open: false,
		});
	};	

	render() {
		return (
			<Snackbar
				open={this.state.open}
				message="Successfully logged out."
				autoHideDuration={4000}
				onRequestClose={this.handleRequestClose}
			/>
		);
	}
}