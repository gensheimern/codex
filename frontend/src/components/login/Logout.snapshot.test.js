import React from 'react';
import renderer from'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import Logout from './Logout';

describe('Logout Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <React.Fragment>
                    <Logout />
                </React.Fragment>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});