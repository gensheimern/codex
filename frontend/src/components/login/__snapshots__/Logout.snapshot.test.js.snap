// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Snapshot renders 1`] = `
<div
  className="static-modal"
>
  <div
    className="modal"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
        role="document"
      >
        <div
          className="modal-header"
        >
          <strong>
            Logout
          </strong>
        </div>
        <div
          className="modal-body"
        >
          Logout successful!
        </div>
        <div
          className="modal-footer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
            >
              Login
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
